### PR TITLE
Python imports analysis

### PR DESCRIPTION
**Pull request by GirZ0n**
**Changes**:
- Added python support.
- Added new methods: `Project.extractKtElementsOfType` and `Project.extractPyElementsOfType`.
- Added the `filePredicate` argument to the `Project.extractElementsOfType` method. Also replaced all use of this method with `Project.extractKtElementsOfType` in the Kotlin analysis.
- Added a new file predicate: `VirtualFile.isPythonRelatedFile`.
- Added Python dependencies analysis: 
  - `ImportStatementPsiAnalyzer`
  - `FromImportStatementPsiAnalyzer`
  - `ImportStatementsAnalysisExecutor`
  - `PythonDependenciesAnalysisRunner`
- Added tests for:
  - `ImportStatementPsiAnalyzer`
  - `FromImportStatementPsiAnalyzer`
  - `ImportStatementsAnalysisExecutor`

### CODE REVIEW

File: python-analysis-dependencies/src/main/kotlin/org/jetbrains/research/ml/pythonAnalysis/FromImportStatementPsiAnalyzer.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * Analyzer for [from import statement][PyFromImportStatement]. Analysis consists of fully qualified
+ * name extraction.
+ */
+object FromImportStatementPsiAnalyzer : PsiAnalyzer<PyFromImportStatement, List<String>> {
+
+    /** Get fully qualified name of given [from import statement][PyFromImportStatement]. */
+    override fun analyze(psiElement: PyFromImportStatement): List<String> {
+        return when (psiElement.relativeLevel) {
+            // Processing absolute import
+            0 -> {
```
Comment (nbirillo) [inline python-analysis-dependencies/src/main/kotlin/org/jetbrains/research/ml/pythonAnalysis/FromImportStatementPsiAnalyzer.kt:17]: Could you please add some const for this number?

File: python-analysis-dependencies/src/main/kotlin/org/jetbrains/research/ml/pythonAnalysis/FromImportStatementPsiAnalyzer.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                when (psiElement.isStarImport) {
```
Comment (nbirillo) [inline python-analysis-dependencies/src/main/kotlin/org/jetbrains/research/ml/pythonAnalysis/FromImportStatementPsiAnalyzer.kt:18]: Maybe here is it better to use if operator?

File: python-analysis-dependencies/src/main/kotlin/org/jetbrains/research/ml/pythonAnalysis/FromImportStatementPsiAnalyzer.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                    true -> psiElement.importSourceQName?.toString()?.let { listOf(it) } ?: emptyList()
+                    false -> psiElement.fullyQualifiedObjectNames
+                }
+            }
+            // Processing relative import
+            else -> {
+                val importSourceQName = psiElement.resolveImportSource()?.getQName()?.toString()
+
+                when (psiElement.isStarImport) {
```
Comment (nbirillo) [inline python-analysis-dependencies/src/main/kotlin/org/jetbrains/research/ml/pythonAnalysis/FromImportStatementPsiAnalyzer.kt:27]: And here too

**Review by nbirillo — CHANGES_REQUESTED** [review]
Thank you a lot for the many tests! @annav1asova Could you please review this PR too?

**Review by nbirillo — APPROVED** [review]
Great! Thank you!

**Review by annav1asova — COMMENTED** [review]
Sorry for being late, but everything looks great, thank you!

**Pull request merged**